### PR TITLE
CLOUDSTACK-9140: Testcase to verify if Dedicated cluster is used for virtual routers that belong to non dedicated account

### PR DESCRIPTION
**Pull request by pritisarap12**

### CODE REVIEW

**Comment by pritisarap12** [comment]
Testcase Result:

Dedicated cluster and router allocation ... === TestName: test_01_dedicated_cluster_allocation | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 302.277s

OK

File: test/integration/component/maint/testpath_disable_enable_zone.py [hunk 1]
```diff
@@ -1691,3 +1696,159 @@ def test_01_disable_enable_host(self):
...
+            self.userapiclient_1,
+            self.testdata["small"],
+            templateid=self.template.id,
+            accountid=self.account_1.name,
+            domainid=self.account_1.domainid,
+            serviceofferingid=self.service_offering.id,
+            zoneid=self.zone.id,
+            mode=self.zone.networktype
+        )
+
```
Comment (sanju1010) [inline test/integration/component/maint/testpath_disable_enable_zone.py]: Can you please add few verification steps to make sure that vm and VR are being deployed on dedicated cluster when deploying with dedicated account?

File: test/integration/component/maint/testpath_disable_enable_zone.py [hunk 2]
```diff
@@ -1691,3 +1696,159 @@ def test_01_disable_enable_host(self):
...
+        # Step 4
+        routerList = list_routers(self.apiclient,
+                                  clusterid=self.clusters[0].id,
+                                  networkid=self.vm_1.nic[0].networkid
+                                  )
+        self.assertEqual(
```
Comment (sanju1010) [inline test/integration/component/maint/testpath_disable_enable_zone.py]: Add similar validation for vm_1 as well

File: test/integration/component/maint/testpath_disable_enable_zone.py (hunk 1 above)
Comment (pritisarap12) [inline test/integration/component/maint/testpath_disable_enable_zone.py]: Added verification steps:
While a cluster gets dedicated to an account a default affinity group gets created, 
then deploy VM using that affinity group to ensure that the VM is deployed on dedicated cluster.

**Comment by pritisarap12** [comment]
Testcase result after modifications:

Dedicated cluster and router allocation ... === TestName: test_01_dedicated_cluster_allocation | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 278.037s

OK

**Comment by shwetaag** [comment]
did code walk through. LGTM

**Pull request merged**